etl/steps/export/s3/co2_data/latest/owid_co2.py

```diff
@@ -0,0 +1,125 @@
+    # * "iso_code", which is the ISO code (as a string), if it exists.
+    # * "data", which is a list of dictionaries, one per year.
+    #   Each dictionary contains "year" as the first item, followed by all other non-nan indicator values for that year.
+    for country in sorted(set(tb["country"])):
+        # Initialize output dictionary for current country.
+        output_dict[country] = {}
+
+        # If there is an ISO code for this country, add it as a new item of the dictionary.
+        iso_code = tb[tb["country"] == country].iloc[0]["iso_code"]
+        if not pd.isna(iso_code):
+            output_dict[country]["iso_code"] = iso_code
+
+        # Create the data dictionary for this country.
+        dict_country = tb[tb["country"] == country].drop(columns=["country", "iso_code"]).to_dict(orient="records")
+        # Remove all nans.
+        data_country = [
+            {indicator: value for indicator, value in d_year.items() if not pd.isna(value)} for d_year in dict_country
+        ]
+        output_dict[country]["data"] = data_country
+
+    # Write dictionary to file as a big json object.
+    with open(output_path, "w") as file:
+        file.write(json.dumps(output_dict, indent=4))
+
+
+def prepare_and_save_outputs(tb: Table, codebook: Table, temp_dir_path: Path) -> None:
+    # Create a csv file.
+    log.info("Creating csv file.")
+    pd.DataFrame(tb).to_csv(temp_dir_path / "owid-co2-data.csv", index=False, float_format="%.3f")
+
+    # Create a json file.
+    log.info("Creating json file.")
+    save_data_to_json(tb, temp_dir_path / "owid-co2-data.json")
+
+    # Create an excel file.
+    log.info("Creating excel file.")
+    with pd.ExcelWriter(temp_dir_path / "owid-co2-data.xlsx") as writer:
+        tb.to_excel(writer, sheet_name="Data", index=False, float_format="%.3f")
+        codebook.to_excel(writer, sheet_name="Metadata")
+
+
+def run(dest_dir: str) -> None:
+    #
+    # Load data.
+    #
+    # Load the owid_co2 emissions dataset from garden, and read its main table.
+    ds_gcp = paths.load_dataset("owid_co2")
+    tb = ds_gcp.read("owid_co2")
+    codebook = ds_gcp.read("owid_co2_codebook")
+
+    #
+    # Save outputs.
+    #
+    # Create a temporary directory for all files to be committed.
+    with tempfile.TemporaryDirectory() as temp_dir:
+        temp_dir_path = Path(temp_dir)
+
+        prepare_and_save_outputs(tb, codebook=codebook, temp_dir_path=temp_dir_path)
+
+        # Initialise S3 client.
+        s3 = S3()
+        for file_name in tqdm(["owid-co2-data.csv", "owid-co2-data.xlsx", "owid-co2-data.json"]):
+            # Path to local file.
+            local_file = temp_dir_path / file_name
+            # Path (within bucket) to S3 file.
+            s3_file = Path("data/co2") / file_name
+            tqdm.write(f"Uploading file {local_file} to S3 bucket {S3_BUCKET_NAME} as {s3_file}.")
+            # Upload and make public each of the files.
+            s3.upload_to_s3(
```

Comment:
I'm thinking about what should be the default automation for export steps. Should we run them all every time, just like the rest of the ETL? It would only work if proper env variables are set in production.

We could also do it by creating a script similar to `update-monkeypox.sh`. Or just keep doing it manually.